api/src/main/java/org/pmiops/workbench/cohorts/CohortService.java

```diff
@@ -68,11 +89,36 @@ public CohortService(
         .build();
   }
 
-  public TerraJob submitGenomicsCohortExtractionJob(String workspaceNamespace, String workspaceName)
+  private List<String> getPersonIds(Long cohortId) {
+    String cohortDefinition = cohortDao.findOne(cohortId).getCriteria();
+
+    final SearchRequest searchRequest = new Gson().fromJson(cohortDefinition, SearchRequest.class);
+
+    final QueryJobConfiguration participantIdQuery =
+        cohortQueryBuilder.buildParticipantIdQuery(new ParticipantCriteria(searchRequest));
+
+    return Streams.stream(
+            bigQueryService
+                .executeQuery(bigQueryService.filterBigQueryConfig(participantIdQuery))
+                .getValues())
+        .map(personId -> personId.get(0).getValue().toString())
+        .collect(Collectors.toList());
+  }
+
+  public TerraJob submitGenomicsCohortExtractionJob(DbWorkspace workspace, Long cohortId)
       throws ApiException {
     WorkbenchConfig.WgsCohortExtractionConfig cohortExtractionConfig =
         workbenchConfigProvider.get().wgsCohortExtraction;
 
+    FirecloudWorkspace fcWorkspace = fireCloudService.getWorkspace(workspace).get().getWorkspace();
+    String extractionUuid = UUID.randomUUID().toString();
+    String filename = extractionUuid + "_person_ids.txt";
+    Blob personIdsFile =
+        cloudStorageService.writeFile(
```

Comment:
I don't love putting this into user-space, since they can potentially muck with it while the extraction is running. It's probably a fine approach for now but we may want a follow-up task to investigate